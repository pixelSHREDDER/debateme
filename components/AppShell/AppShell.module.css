.container {
  background-color: light-dark(var(--mantine-primary-color-3), var(--mantine-primary-color-8));
  border-bottom: rem(1px) solid light-dark(var(--mantine-color-gray-4), var(--mantine-color-dark-4));
  height: rem(56px);
  display: flex;
  justify-content: space-between;
  align-items: center;

  a {
    color: light-dark(var(--mantine-color-purple-6), var(--mantine-color-purple-2));
  }
}

.title {
  color: light-dark(var(--mantine-color-black), var(--mantine-color-white));
  font-size: rem(100px);
  font-weight: 900;
  letter-spacing: rem(-2px);

  @media (max-width: $mantine-breakpoint-md) {
    font-size: rem(50px);
  }
}

.link {
  text-transform: uppercase;
  font-size: var(--mantine-font-size-xs);
  color: light-dark(var(--mantine-color-gray-6), var(--mantine-color-dark-1));
  padding: rem(20px) var(--mantine-spacing-sm);
  font-weight: 700;
  transition:
    border-color 100ms ease,
    color 100ms ease;

  @mixin hover {
    color: light-dark(var(--mantine-color-black), var(--mantine-color-white));
    text-decoration: none;
  }

  &[data-active] {
    color: light-dark(var(--mantine-color-black), var(--mantine-color-white));
    border-bottom-color: var(--mantine-primary-color-6);
  }
}

.user {
  border-inline-end: 1px solid var(--mantine-color-black);
  padding-inline-end: var(--mantine-spacing-xs);
}

.headerLink {
  padding: rem(7px) var(--mantine-spacing-sm);
  border-bottom: rem(2px) solid transparent;

    &:not(:last-child) {
      border-inline-end: 1px solid light-dark(var(--mantine-color-gray-4), var(--mantine-color-dark-4));
      padding-inline-end: var(--mantine-spacing-xs);
    }
}
