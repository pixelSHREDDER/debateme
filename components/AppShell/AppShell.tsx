'use client'

import cx from 'clsx'
import {
  AppShell as MantineAppShell,
  Burger,
  Group,
  Anchor,
  Stack,
  Image,
} from '@mantine/core'
import { useDisclosure } from '@mantine/hooks'
import classes from './AppShell.module.css'
import { useUser } from '@auth0/nextjs-auth0/client'
import { useState } from 'react'
import { ColorSchemeToggle } from '../ColorSchemeToggle/ColorSchemeToggle'
import { usePathname } from 'next/navigation'

const links = [
  { link: '/', label: 'Home', logOutHide: false },
  { link: '/debate', label: 'Debates', logOutHide: true },
]

export default function AppShell({ children }: { children: React.ReactNode }) {
  const pathname = usePathname()
  const { error, isLoading, user } = useUser()
  const [opened, { toggle }] = useDisclosure(false)
  const [active, setActive] = useState(links[0].link)
  const visibleLinks = user ? links : links.filter(link => !link.logOutHide)

  const items = (isHeader: boolean) => visibleLinks.map((link) => (
    <Anchor<'a'>
      href={link.link}
      key={link.label}
      className={cx(classes.link, {[classes.headerLink]: !!isHeader})}
      data-active={active === link.link || undefined}
      onClick={() => setActive(link.link)}
    >
      {link.label}
    </Anchor>
  ))

  const userChunk = () => {
    if (isLoading) return 'loading user....'
    if (error) return 'error'
    if (user) {
      return (
        <>
          <Anchor<'a'> href="/user/profile" className={classes.user}>Hi, {user.name}!</Anchor><Anchor<'a'> href="/api/auth/logout" className={classes.link}>Logout</Anchor>
        </>
      )
    }
    return <Anchor<'a'> href="/api/auth/login" className={classes.link}>Login</Anchor>
  }

  return (
    <MantineAppShell
      header={{ height: 60 }}
      navbar={{ width: 300, breakpoint: 'sm', collapsed: { desktop: true, mobile: !opened } }}
      padding={pathname === '/' ? 0 : "md"}
    >
      <MantineAppShell.Header className={classes.header}>
        <Group h="100%" justify="space-between" w="100%" px="md" className={classes.container}>
          <Image
            height={28}
            src="/debate-me-logo.svg"
					  alt="Debate.me logo" />
          <Group gap={5} visibleFrom="sm">
            {items(true)}
          </Group>
          <Group justify="right" visibleFrom="sm">
            <ColorSchemeToggle />
            {userChunk()}
          </Group>
          <Burger opened={opened} onClick={toggle} hiddenFrom="sm" size="sm" />
        </Group>
      </MantineAppShell.Header>
      <MantineAppShell.Navbar py="md" px={8}>
        <MantineAppShell.Section grow>
          <Stack gap={20}>
            {items(false)}
          </Stack>
        </MantineAppShell.Section>
        <MantineAppShell.Section>
          <Group align="end" justify="space-between">
            <ColorSchemeToggle />
            {userChunk()}
          </Group>
        </MantineAppShell.Section>
      </MantineAppShell.Navbar>
      <MantineAppShell.Main>
        {children}
      </MantineAppShell.Main>
    </MantineAppShell>
  )
}
