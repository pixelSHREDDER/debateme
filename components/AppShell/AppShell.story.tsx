import type { Meta } from '@storybook/react'
import AppShell from './AppShell'
import React from 'react'

const meta: Meta<typeof AppShell> = {
  title: 'App Shell',
}

export default meta

export const Usage = () => <AppShell><p>App Shell main content</p></AppShell>
