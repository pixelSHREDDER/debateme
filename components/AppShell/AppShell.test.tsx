import { describe, expect, test } from 'vitest'
import { render } from '@testing-library/react'
import ProviderWrapper from '@/tests/helpers/ProviderWrapper'
import AppShell from './AppShell'
import React from 'react'

describe('App Shell component', () => {
  test('renders correctly', () => {
    const { container } = render(
      <AppShell>
        <p>test main content</p>
      </AppShell>,
      { wrapper: ProviderWrapper })
    expect(container).toMatchSnapshot()
  })
})
