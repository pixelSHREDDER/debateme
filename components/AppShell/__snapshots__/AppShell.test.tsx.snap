// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Shell component > renders correctly 1`] = `
<div>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: mocked, sans-serif;--mantine-font-family-headings: mocked, sans-serif;--mantine-primary-color-filled: var(--mantine-color-marble-filled);--mantine-primary-color-filled-hover: var(--mantine-color-marble-filled-hover);--mantine-primary-color-light: var(--mantine-color-marble-light);--mantine-primary-color-light-hover: var(--mantine-color-marble-light-hover);--mantine-primary-color-light-color: var(--mantine-color-marble-light-color);--mantine-primary-color-0: var(--mantine-color-marble-0);--mantine-primary-color-1: var(--mantine-color-marble-1);--mantine-primary-color-2: var(--mantine-color-marble-2);--mantine-primary-color-3: var(--mantine-color-marble-3);--mantine-primary-color-4: var(--mantine-color-marble-4);--mantine-primary-color-5: var(--mantine-color-marble-5);--mantine-primary-color-6: var(--mantine-color-marble-6);--mantine-primary-color-7: var(--mantine-color-marble-7);--mantine-primary-color-8: var(--mantine-color-marble-8);--mantine-primary-color-9: var(--mantine-color-marble-9);--mantine-color-gray-0: #f0f3fd;--mantine-color-gray-1: #e0e3ee;--mantine-color-gray-2: #bfc4d9;--mantine-color-gray-3: #9ca4c5;--mantine-color-gray-4: #7e88b3;--mantine-color-gray-5: #6b77a9;--mantine-color-gray-6: #616ea5;--mantine-color-gray-7: #515d91;--mantine-color-gray-8: #475283;--mantine-color-gray-9: #3a4775;--mantine-color-marble-0: #faf7ea;--mantine-color-marble-1: #efecde;--mantine-color-marble-2: #dbd7bf;--mantine-color-marble-3: #c8c29d;--mantine-color-marble-4: #b7af81;--mantine-color-marble-5: #aca36e;--mantine-color-marble-6: #a79d62;--mantine-color-marble-7: #918951;--mantine-color-marble-8: #817945;--mantine-color-marble-9: #706836;--mantine-color-purple-0: #f6eeff;--mantine-color-purple-1: #e7daf7;--mantine-color-purple-2: #cab1ea;--mantine-color-purple-3: #ad86dd;--mantine-color-purple-4: #9562d2;--mantine-color-purple-5: #854bcb;--mantine-color-purple-6: #7d3ec9;--mantine-color-purple-7: #6b31b2;--mantine-color-purple-8: #5f2aa0;--mantine-color-purple-9: #52228d;}:root[data-mantine-color-scheme="dark"]{--mantine-color-anchor: var(--mantine-color-marble-4);--mantine-color-gray-light: rgba(97, 110, 165, 0.15);--mantine-color-gray-light-hover: rgba(97, 110, 165, 0.2);--mantine-color-gray-outline-hover: rgba(126, 136, 179, 0.05);--mantine-color-marble-text: var(--mantine-color-marble-4);--mantine-color-marble-filled: var(--mantine-color-marble-8);--mantine-color-marble-filled-hover: var(--mantine-color-marble-9);--mantine-color-marble-light: rgba(167, 157, 98, 0.15);--mantine-color-marble-light-hover: rgba(167, 157, 98, 0.2);--mantine-color-marble-light-color: var(--mantine-color-marble-3);--mantine-color-marble-outline: var(--mantine-color-marble-4);--mantine-color-marble-outline-hover: rgba(183, 175, 129, 0.05);--mantine-color-purple-text: var(--mantine-color-purple-4);--mantine-color-purple-filled: var(--mantine-color-purple-8);--mantine-color-purple-filled-hover: var(--mantine-color-purple-9);--mantine-color-purple-light: rgba(125, 62, 201, 0.15);--mantine-color-purple-light-hover: rgba(125, 62, 201, 0.2);--mantine-color-purple-light-color: var(--mantine-color-purple-3);--mantine-color-purple-outline: var(--mantine-color-purple-4);--mantine-color-purple-outline-hover: rgba(149, 98, 210, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-anchor: var(--mantine-color-marble-6);--mantine-color-gray-light: rgba(97, 110, 165, 0.1);--mantine-color-gray-light-hover: rgba(97, 110, 165, 0.12);--mantine-color-gray-outline-hover: rgba(97, 110, 165, 0.05);--mantine-color-marble-text: var(--mantine-color-marble-filled);--mantine-color-marble-filled: var(--mantine-color-marble-6);--mantine-color-marble-filled-hover: var(--mantine-color-marble-7);--mantine-color-marble-light: rgba(167, 157, 98, 0.1);--mantine-color-marble-light-hover: rgba(167, 157, 98, 0.12);--mantine-color-marble-light-color: var(--mantine-color-marble-6);--mantine-color-marble-outline: var(--mantine-color-marble-6);--mantine-color-marble-outline-hover: rgba(167, 157, 98, 0.05);--mantine-color-purple-text: var(--mantine-color-purple-filled);--mantine-color-purple-filled: var(--mantine-color-purple-6);--mantine-color-purple-filled-hover: var(--mantine-color-purple-7);--mantine-color-purple-light: rgba(125, 62, 201, 0.1);--mantine-color-purple-light-hover: rgba(125, 62, 201, 0.12);--mantine-color-purple-light-color: var(--mantine-color-purple-6);--mantine-color-purple-outline: var(--mantine-color-purple-6);--mantine-color-purple-outline-hover: rgba(125, 62, 201, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <style
    data-mantine-styles="inline"
  >
    :root{--app-shell-navbar-width:calc(18.75rem * var(--mantine-scale));--app-shell-navbar-offset:calc(18.75rem * var(--mantine-scale));--app-shell-header-height:calc(3.75rem * var(--mantine-scale));--app-shell-header-offset:calc(3.75rem * var(--mantine-scale));--app-shell-padding:var(--mantine-spacing-md);}@media(min-width: 48em){:root{--app-shell-navbar-transform:translateX(calc(var(--app-shell-navbar-width) * -1));--app-shell-navbar-transform-rtl:translateX(var(--app-shell-navbar-width));--app-shell-navbar-offset:0px !important;}}@media(max-width: 47.99375em){:root{--app-shell-navbar-width:100%;--app-shell-navbar-offset:0px;--app-shell-navbar-transform:translateX(calc(var(--app-shell-navbar-width) * -1));--app-shell-navbar-transform-rtl:translateX(var(--app-shell-navbar-width));}}
  </style>
  <div
    class="m_89ab340 mantine-AppShell-root"
    data-resizing="true"
    style="--app-shell-transition-duration: 200ms; --app-shell-transition-timing-function: ease;"
  >
    <header
      class="m_3b16f56b mantine-AppShell-header right-scroll-bar-position _header_901b6f"
      data-with-border="true"
      style="--app-shell-header-z-index: 100;"
    >
      <div
        class="_container_901b6f m_4081bf90 mantine-Group-root"
        style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: space-between; --group-wrap: wrap; padding-inline: var(--mantine-spacing-md); width: 100%; height: 100%;"
      >
        <img
          alt="Debate.me logo"
          class="m_9e117634 mantine-Image-root"
          height="28"
          src="/debate-me-logo.svg"
        />
        <div
          class="m_4081bf90 mantine-Group-root mantine-visible-from-sm"
          style="--group-gap: calc(0.3125rem * var(--mantine-scale)); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
        >
          <a
            class="mantine-focus-auto m_849cf0da _link_901b6f _headerLink_901b6f m_b6d8b162 mantine-Text-root mantine-Anchor-root"
            data-active="true"
            data-underline="hover"
            href="/"
          >
            Home
          </a>
        </div>
        <div
          class="m_4081bf90 mantine-Group-root mantine-visible-from-sm"
          style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: right; --group-wrap: wrap;"
        >
          <div
            class="m_4081bf90 mantine-Group-root"
            style="--group-gap: calc(0.0625rem * var(--mantine-scale)); --group-align: center; --group-justify: center; --group-wrap: wrap;"
          >
            <button
              aria-label="Toggle color scheme"
              class="mantine-focus-auto mantine-active m_8d3f4000 mantine-ActionIcon-root m_87cf2631 mantine-UnstyledButton-root"
              data-size="lg"
              data-variant="default"
              style="--ai-size: var(--ai-size-lg); --ai-bg: var(--mantine-color-default); --ai-hover: var(--mantine-color-default-hover); --ai-color: var(--mantine-color-default-color); --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-default-border);"
              type="button"
            >
              <span
                class="m_8d3afb97 mantine-ActionIcon-icon"
              >
                <svg
                  class="tabler-icon tabler-icon-sun _icon_c853fd _light_c853fd"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 12m-4 0a4 4 0 1 0 8 0a4 4 0 1 0 -8 0"
                  />
                  <path
                    d="M3 12h1m8 -9v1m8 8h1m-9 8v1m-6.4 -15.4l.7 .7m12.1 -.7l-.7 .7m0 11.4l.7 .7m-12.1 -.7l-.7 .7"
                  />
                </svg>
                <svg
                  class="tabler-icon tabler-icon-moon _icon_c853fd _dark_c853fd"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z"
                  />
                </svg>
              </span>
            </button>
          </div>
          loading user....
        </div>
        <button
          class="mantine-focus-auto m_fea6bf1a mantine-Burger-root m_87cf2631 mantine-UnstyledButton-root mantine-hidden-from-sm"
          data-size="sm"
          style="--burger-size: var(--burger-size-sm);"
          type="button"
        >
          <div
            class="m_d4fb9cad mantine-Burger-burger"
            data-reduce-motion="true"
          />
        </button>
      </div>
    </header>
    <nav
      class="m_45252eee mantine-AppShell-navbar"
      data-with-border="true"
      style="--app-shell-navbar-z-index: calc(100 + 1); padding-inline: calc(0.5rem * var(--mantine-scale)); padding-block: var(--mantine-spacing-md);"
    >
      <div
        class="m_6dcfc7c7 mantine-AppShell-section"
        data-grow="true"
      >
        <div
          class="m_6d731127 mantine-Stack-root"
          style="--stack-gap: calc(1.25rem * var(--mantine-scale)); --stack-align: stretch; --stack-justify: flex-start;"
        >
          <a
            class="mantine-focus-auto m_849cf0da _link_901b6f m_b6d8b162 mantine-Text-root mantine-Anchor-root"
            data-active="true"
            data-underline="hover"
            href="/"
          >
            Home
          </a>
        </div>
      </div>
      <div
        class="m_6dcfc7c7 mantine-AppShell-section"
      >
        <div
          class="m_4081bf90 mantine-Group-root"
          style="--group-gap: var(--mantine-spacing-md); --group-align: end; --group-justify: space-between; --group-wrap: wrap;"
        >
          <div
            class="m_4081bf90 mantine-Group-root"
            style="--group-gap: calc(0.0625rem * var(--mantine-scale)); --group-align: center; --group-justify: center; --group-wrap: wrap;"
          >
            <button
              aria-label="Toggle color scheme"
              class="mantine-focus-auto mantine-active m_8d3f4000 mantine-ActionIcon-root m_87cf2631 mantine-UnstyledButton-root"
              data-size="lg"
              data-variant="default"
              style="--ai-size: var(--ai-size-lg); --ai-bg: var(--mantine-color-default); --ai-hover: var(--mantine-color-default-hover); --ai-color: var(--mantine-color-default-color); --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-default-border);"
              type="button"
            >
              <span
                class="m_8d3afb97 mantine-ActionIcon-icon"
              >
                <svg
                  class="tabler-icon tabler-icon-sun _icon_c853fd _light_c853fd"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 12m-4 0a4 4 0 1 0 8 0a4 4 0 1 0 -8 0"
                  />
                  <path
                    d="M3 12h1m8 -9v1m8 8h1m-9 8v1m-6.4 -15.4l.7 .7m12.1 -.7l-.7 .7m0 11.4l.7 .7m-12.1 -.7l-.7 .7"
                  />
                </svg>
                <svg
                  class="tabler-icon tabler-icon-moon _icon_c853fd _dark_c853fd"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z"
                  />
                </svg>
              </span>
            </button>
          </div>
          loading user....
        </div>
      </div>
    </nav>
    <main
      class="m_8983817 mantine-AppShell-main"
    >
      <p>
        test main content
      </p>
    </main>
  </div>
</div>
`;
